plugins/newsdownloader.koplugin/epubdownloadbackend.lua

```diff
@@ -24,6 +24,50 @@ local max_redirects = 5; --prevent infinite redirects
 local TIMEOUT_CODE = "timeout" -- from socket.lua
 local MAXTIME_CODE = "maxtime reached" -- from sink_table_with_maxtime
 
+-- filter HTML using CSS selector
+local function filter(text, element)
+    local htmlparser = require("htmlparser")
+    local root = htmlparser.parse(text)
+    local filtered = ""
```

Comment:
You could just use `filtered = nil` - and use it instead of `stop` below. It would become more natural.
Just 2 `if filtered then break end` and at the end `if not filtered then return text end`